### ISSUE TITLE
"MaxListenersExceeded" warning

### ISSUE THREAD

**Issue by zachweintraub** [zachweintraub/scum#1]
Locally or deployed, the server logs this warning after playing for a short time:
```
MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 GAME_UPDATED listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
```
The game's performance also seems to slow gradually. This may or may not be related, but it feels like it is.
Note: the above warning was triggered during a game with only 5 players, so it seems something funny is definitely going on.

**Comment by ckeyes88** [comment]
Gathering some resources:
- https://stackoverflow.com/questions/59887163/how-can-i-configure-the-apollo-server-pubsub-eventemitter-to-throw-or-log-an-err
- This might be an easy alternative to the built in pubsub: https://github.com/m19c/graphql-firestore-subscriptions
- https://www.apollographql.com/docs/apollo-server/data/subscriptions/#pubsub-implementations

**Comment by zachweintraub** [comment]
Nice thanks Chip!

I was looking at the built in pubsub and it looks like it removes event listeners so not sure what's going on. Would really like to _understand_ why the game slows down progressively.

Good find on the firestore pubsub - I feel like that's probably easier (and free-er?) to set up versus Redis or RabbitMQ. I'm also still interested in a DIY alternative just for understanding all of this better. But firestore is probably a quicker fix for getting the game performant in the short term.

**Comment by zachweintraub** [comment]
The issue was actually in the client. I stuck a console log in the `PubSub`'s `publish` method to print out the current number of subscriptions. It turned out that `subscribeToMore` was being called in the `Game` component and getting triggered 3 times per game update, so the number of active subscriptions was growing continuously throughout gameplay.

Found someone else with the same issue [here](https://github.com/apollographql/react-apollo/issues/2656).

To fix, I queried for the game one level higher and passed the subscribe function in as a prop. Then, I called it in a `useEffect`.
